discovery_child_development/analysis/baseline_model.py

```diff
@@ -0,0 +1,423 @@
+    model_path: str = MODEL_PATH,
+):
+    valid_inputs = ["majority_combination", "most_probable"]  # "majority_label",
+
+    if model_type not in valid_inputs:
+        raise ValueError(
+            f"Invalid input. Expected one of {valid_inputs}, got '{model_type}'"
+        )
+
+    # Load the data
+    training_data_filename = (
+        f"openalex_data_{concept_ids}_year-2019-2020-2021-2022-2023_train.csv"
+    )
+
+    openalex_data = S3.download_obj(
+        s3_bucket,
+        path_from=f"{input_path}{training_data_filename}",
+        download_as="dataframe",
+        kwargs_reading={"index_col": 0},
+    )
+
+    if wandb_run:
+        # Initialize a wandb run and log score threshold with wandb
+        run = wandb.init(
+            project="ISS supervised ML",
+            job_type="Baseline modeling",
+            save_code=True,
+            config={
+                "baseline_model_type": model_type,
+                "score_threshold": score_threshold,
+            },
+        )
+        # We will use set the wandb description to be the dataset name (which includes details of concepts and years)
+        # - we set the description, and not the name, as there is a limit on the length of artifact names.
+        # wandb artifact names also cannot include "|" so we replace this.
+        wandb_name = training_data_filename.replace("|", "_")
+        # add reference to this data in wandb
+        wb.add_ref_to_data(
+            run,
+            "openalex_train_data_raw",
+            wandb_name,
+            S3_BUCKET,
+            training_data_filename,
+        )
+
+    # Filter the data using a score threshold (0.3 is the threshold used by OpenAlex)
+    openalex_data_wide = (
+        openalex_data[openalex_data["score"] >= score_threshold]
+        # Squash sub-categories into one tuple per work (rather than one row per sub-category per work)
+        .groupby(["openalex_id", "text"])["sub_category"]
+        .agg(lambda x: tuple(set(x)))
+        .reset_index()
+    )
+    # Set the index - useful later for creating training/validation split
+    openalex_data_wide = openalex_data_wide.set_index("openalex_id")
+
+    top_combinations, most_common_combination = find_most_frequent_labels(
+        openalex_data_wide, label_col="sub_category", head=20
+    )
+
+    # Load embeddings
+    embeddings = S3.download_obj(
+        s3_bucket,
+        path_from=vectors_filepath,
+        download_as="dataframe",
+    )
+
+    embeddings = embeddings.set_index("openalex_id")
+
+    openalex_data_wide = openalex_data_wide.join(
+        embeddings, on="openalex_id", how="left"
+    )
+
+    Y, mlb = classification_utils.add_binarise_labels(
+        openalex_data_wide, label_column="sub_category", not_valid_label=None
+    )
+
+    # Split IDs into random train and test subsets
+    unique_ids = openalex_data_wide.index.unique()
+
+    train_ids, val_ids = train_test_split(unique_ids, test_size=0.1, random_state=SEED)
```

Comment:
Just noting that you're splitting your dataset by IDs - I wonder what are the pros/cons for splitting not by IDs but by ID-label pairs (as you can have more than one label per ID)

Here, however, I guess we come back to the point about how you've encoded the labels. Again, I'd suggest encoding the labels not as combinations of sub-categories but simply each label as a single sub-category (see previous comment)

Reply:
That's a really good point. My first thought is that if the labels are correlated in some way (eg documents with "label 1" are also likely to be tagged with "label 2", whereas documents with "label 3" are likely to have no other labels) it would be important to make sure that no IDs straddle the training and validation sets, so that the model doesn't already know that an ID in the validation set has been tagged with "label 1".

Reply:
Yeah, I guess let's keep the IDs separate across test and train sets as you've done so far.